### PR TITLE
Add support for images and actions no matter what the app is doing.

### PR DESCRIPTION
**Pull request by JBassett**
Needs https://github.com/home-assistant/mobile-apps-fcm-push/pull/2 to function.

Still need to think about backwards compatibility...

### CODE REVIEW

**Comment by balloob** [comment]
In what way wouldn't it be backwards compatible ? Actionable notifications have not been released yet.

**Comment by JBassett** [comment]
Because of where we are pulling the title and body of the notification from.  In order to get the data delivered to the `onMessageReceived` function of our code we need to ensure that it is considered a data message.  This means we can't pass the title and message in the normal locations.  If we do pass them normally then we don't get the chance to process the data and the device just displays the notification.
https://firebase.google.com/docs/cloud-messaging/android/receive#handling_messages.

File: app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt [hunk 1]
```diff
@@ -60,20 +67,28 @@ class MessagingService : FirebaseMessagingService() {
...
 
-        remoteMessage.notification?.let {
-            if (it.body == "request_location_update") {
+            if (!it.containsKey(MESSAGE)) {
+                Log.e(TAG, "Message missing from notification.")
+                return
+            }
+
+            val title = it[TITLE]
+            val message = it[MESSAGE]!!
```
Comment (marciogranzotto) [inline app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt]: Why break the app if it's `null`?

```suggestion
            val message = it[MESSAGE] ?: ""
```

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt]: The message is a required field from HA.  If it doesn't make it to the application we have a problem.

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/notify/__init__.py#L45

**Comment by balloob** [comment]
Oh I see the backwards compat issue now. It means that older apps won't be able to receive notifications, because it's in the data channel and that's not supported.

Could we push an update with this PR merged, and then give everyone a week to update their apps before we upgrade the push notifications cloud handling ?

**Comment by JBassett** [comment]
The changes I made to the cloud function adds the new functionality in a new URL/function.  That way we can keep the old functionality around as long as we need.  Because I changed the `push_url` in the app it does need the new function in firebase first.

**Comment by balloob** [comment]
Ok!

**Pull request merged**